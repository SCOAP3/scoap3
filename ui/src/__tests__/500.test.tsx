import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { useRouter } from "next/navigation";

import ServerErrorPage from "@/pages/500";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

const mockPush = jest.fn();
(useRouter as jest.Mock).mockImplementation(() => ({
  push: mockPush,
}));

describe("ServerErrorPage", () => {
  it("renders error page", () => {
    const { container } = render(<ServerErrorPage />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText("Something went wrong")).toBeInTheDocument();
    expect(screen.getByTestId("go-back")).toBeInTheDocument();
  });

  it("triggers router.push on 'go to home page' button click", () => {
    render(<ServerErrorPage />);

    fireEvent.click(screen.getByTestId("go-back"));

    expect(mockPush).toHaveBeenCalledWith("/");
  });
});
