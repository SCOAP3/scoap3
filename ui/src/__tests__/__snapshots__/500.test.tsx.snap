// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerErrorPage renders error page 1`] = `
<div>
  <div
    class="error-page"
  >
    <span
      aria-label="exclamation-circle"
      class="anticon anticon-exclamation-circle"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="exclamation-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
        <path
          d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
        />
      </svg>
    </span>
    <h1>
      Something went wrong
    </h1>
    <p>
      Please try again later or

      <a
        data-testid="go-back"
        type="button"
      >
        go to home page
      </a>
    </p>
  </div>
</div>
`;
